utils/fifoqueue/fifoqueue.go

```diff
@@ -83,23 +87,33 @@ func NewFifoQueue(options ...ConstructorOption) (*FifoQueue, error) {
 
 // Push appends the given value to the tail of the queue.
 // If queue capacity is reached, the message is silently dropped.
-func (q *FifoQueue) Push(element interface{}) {
+func (q *FifoQueue) Push(element interface{}) bool {
```

Comment:
I have a strong opinion about invoking callbacks when object is holding a lock, this is error phone and should be avoided. I suggest to align it with `Pop()` implementation and release lock before notifying.